sdks/java/io/azure/src/main/java/org/apache/beam/sdk/io/azure/blobstore/AzureReadableSeekableByteChannel.java

```diff
@@ -44,23 +50,22 @@ public int read(ByteBuffer dst) throws IOException {
     if (closed) {
       throw new ClosedChannelException();
     }
-    if (!dst.hasRemaining()) {
-      return 0;
-    }
 
     int read = 0;
     if (dst.hasArray()) {
       // Stores up to dst.remaining() bytes into dst.array() starting at dst.position().
       // But dst can have an offset with its backing array, hence the + dst.arrayOffset().
       read = inputStream.read(dst.array(), dst.position() + dst.arrayOffset(), dst.remaining());
+      LOG.info("PArray: " + StandardCharsets.UTF_8.decode(dst).toString());
     } else {
       byte[] myarray = new byte[dst.remaining()];
       read = inputStream.read(myarray, 0, myarray.length);
       dst.put(myarray);
+      LOG.info("Array: " + Arrays.toString(myarray));
```

Comment:
I recommend you remove all the LOG.info lines from this class. Because there may be 100s or 1000s of AzureReadableSeekableByteChannel objects in a Beam pipeline, this logging may overwhelm the logging system

Reply:
Got it, will do.